src/modules/QtQuick/Loader.js

```diff
@@ -44,32 +44,41 @@ QmlWeb.registerQmlType({
     if (!this.active) return;
     this.$unload();
 
-    const tree = QmlWeb.engine.loadComponent(fileName);
-    const QMLComponent = QmlWeb.getConstructor("QtQml", "2.0", "Component");
-    const meta = { object: tree, context: this, parent: this };
-    const qmlComponent = new QMLComponent(meta);
-    qmlComponent.$basePath = QmlWeb.engine.extractBasePath(tree.$file);
-    qmlComponent.$imports = tree.$imports;
-    qmlComponent.$file = tree.$file;
-    QmlWeb.engine.loadImports(tree.$imports, qmlComponent.$basePath,
-      qmlComponent.importContextId);
-    const loadedComponent = this.$createComponentObject(qmlComponent, this);
-    this.sourceComponent = loadedComponent;
+    if (fileName) {
```

Comment:
Moving this one level deeper is not needed, just use this check instead:
```js
if (!fileName) {
  this.sourceComponent = null;
  this.$sourceUrl = fileName;
  return;
}
```

Not sure about `$sourceUrl`, perhaps it should be normalized to some falsy value. Or not.